src/main/java/org/radarcns/connect/mongodb/MongoDbWriter.java

```diff
@@ -139,6 +139,8 @@ public void run() {
     private void flushLocalBuffer(List<SinkRecord> localBuffer) {
         localBuffer.stream()
                 .map(KafkaDocument::new)
+                // if schema and/or value is null, skip the record
+                .filter(e -> e.getRecord().valueSchema() != null)
```

Comment:
Can be moved before the `map()`. It would be good to log this as well with warning, since it means that the data cannot be mapped.